zero-ui/src/stores/simulation.ts

```diff
@@ -0,0 +1,86 @@
+import { mutationWithoutValue, mutationWithValue } from "@/graphql/thrs";
+import { context } from "@/graphql/thrs/client";
+
+import { gql, useClientHandle, useQuery } from "@urql/vue";
+import { useIntervalFn } from "@vueuse/core";
+import { OperationResult } from "graphql-ws";
+import { Maybe } from "graphql/jsutils/Maybe";
+import { TypedDocumentNode } from "msw/core/graphql";
+import { defineStore } from "pinia";
+import { computed, ref } from "vue";
+
+type SimulationStatus = {
+  simulation: {
+    status: "running" | "available" | "stepping";
+    time: number;
+  };
+};
+
+const STATUS_QUERY = gql`
+  query SimulationStatus {
+    simulation {
+      status
+      time
+    }
+  }
+`;
+
+export const useSimulationStore = defineStore("simulation", () => {
+  // For some reason the injected client does not come from the component in the outer scope.
+  // This means we need to manually provide the context to each query and mutation.
```

Comment:
Maybe this is because the mutations are defined in the store. Maybe @urql expects the actions to be defined inside the components